File: 14.链表中倒数第k个结点/链表中倒数第k个结点.cpp
/*
struct ListNode {
	int val;
	struct ListNode *next;
	ListNode(int x) :
			val(x), next(NULL) {
	}
};*/
class Solution {
public:
    ListNode* FindKthToTail(ListNode* pListHead, unsigned int k) {
        ListNode* left = pListHead;
        ListNode* right = pListHead;
        if(pListHead==NULL||k<=0){
            return NULL;
        }
        while(k>1){
            if(right->next!=NULL){
                right = right->next;
                k--;
            }else{
                return NULL;
            }
        }
        while(right->next!=NULL){
            left = left->next;
            right = right->next;
        }
        return left;
    }
};

File: 14.链表中倒数第k个结点/链表中倒数第k个结点.java
/*
public class ListNode {
    int val;
    ListNode next = null;

    ListNode(int val) {
        this.val = val;
    }
}*/
public class Solution {
    public ListNode FindKthToTail(ListNode head,int k) {
        if(head==null||k<=0){
            return null;
        }
        ListNode left = head;
        ListNode right = head;
        while(k>1){
            if(right.next!=null){
                right = right.next;
                k--;
            }else{
                return null;
            }
        }
        while(right.next!=null){
            left = left.next;
            right = right.next;
        }
        return left;
    }
}